app/src/main/java/com/example/tasky/itemdetail/presentation/viewmodel/EventDetailViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.tasky.itemdetail.presentation.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.TaskyRoutes
+import com.example.tasky.itemdetail.domain.EventRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class EventDetailViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val repository: EventRepository
+) : ViewModel() {
+
+    var state by mutableStateOf(EventDetailState())
+        private set
+
+    init {
+        val eventId = savedStateHandle.get<String>(TaskyRoutes.EventDetailScreen.EVENT_ID) ?: ""
+        state = state.copy(eventId = eventId)
+        viewModelScope.launch {
```

Comment:
I'd call this only if the eventId from the savedstatehandle is not empty

Or else, you'll search for an empty id

Reply:
Yes, I'd rather use a null value here if the event ID doesn't exist since that's exactly what we use to show something doesn't exist. An empty ID still raises the expectation that it exists